### PR TITLE
remove lodash dep

### PR DESCRIPTION
**Pull request by bsouthga**
This PR removes the lodash dependency, which was really only being used for the `_.find` method. A polyfill has been added.

### CODE REVIEW

File: src/angularjs-dropdown-multiselect.js [hunk 1]
```diff
@@ -117,6 +117,35 @@ directiveModule.directive('ngDropdownMultiselect', ['$filter', '$document', '$co
                     });
                 }
 
+                function deepEqual(x, y) {
+                  if ((typeof x == "object" && x != null) && (typeof y == "object" && y != null)) {
```
Comment (yangchristian) [inline src/angularjs-dropdown-multiselect.js]: Rest of lib conforms to strict `===` comparisons and uses single quotes for strings, should match here.

Also, do you know where this polyfill is from? Lodash's [`_.isEqual()`](https://github.com/lodash/lodash/blob/master/lodash.js#L9470) and the first few google results like [`node-deep-equal`](https://github.com/substack/node-deep-equal) are pretty involved, just want to make sure all our bases are covered.

**Comment by yangchristian** [comment]
Thanks for the equality change. Any thoughts on the `deepEqual()` implementation? For example, I think `for (var prop in x)` iterates all props including prototype props in `x`, but then we don't evaluate prototype props in `y` due to `hasOwnProperty`. Also I think array comparisons would fail.

Lastly, quick style change on the single quote strings would be great. Thanks!

**Comment by bsouthga** [comment]
yeah I've been digging around a bit re: deepEqual, the implementation I dumped in there was a one-off write on my part based on this http://stackoverflow.com/a/25456134. I found https://github.com/substack/node-deep-equal/blob/master/index.js#L43, which is pretty comprehensive and looks like the main difference is comparison of dates + buffers. I think the light weight non-prototype check is ok here, as its mainly being used on the config obj of string keys/values. it seems to work ok on arrays, and I added a length check for the off chance of sparse arrays. thoughts?

**Comment by bsouthga** [comment]
we could also keep this in a separate branch (and use that for CL stuff) if we want to keep this pristine for potential merge back into the original repo...

**Comment by yangchristian** [comment]
Summarizing our slack chat: will go different route of just upgrading lodash in this module and in the other bower deps of our main app to try to get down to a single lodash version (that will be `browserify-shim`med).

**Pull request closed without merging**